src/proxy-animation.js

```diff
@@ -586,27 +1243,47 @@ export class ProxyAnimation {
   }
 
   get finished() {
-    proxyAnimations.get(this).animation.finished;
+    const details = proxyAnimations.get(this);
+    if (!details.timeline)
+       return details.animation.finished;
+
+    if (!details.finishedPromise) {
+      details.finishedPromise = new PromiseWrapper(details);
+      // if (this.playState == 'finished')
+      //   details.finishedPromise.scheduleAsyncFinish();
```

Comment:
What is this commented out code for? Should the finished promise for a finished animation be resolved already?

Reply:
Whoops. Should not have left this in.  If the animation, is already finished, then the promise should be resolved.  Scheduling an async finish was the wrong way to do it however.   If we don't have a finished promise already, then we should not be finished.  updateFinishedState should take care of this.  Will need to change the null resets to new PromiseWrappers.

Reply:
Removed the dead code.